content/4_Gold/PURS.mdx

```diff
@@ -95,6 +92,105 @@ using std::endl;
 using std::vector;
 
 /** A data structure that can answer point update & range minimum queries. */
+template <class T> class MinSegmentTree {
+  private:
+	const T DEFAULT = std::numeric_limits<T>().max();
+
+	int len;
+	int padded_len;
+	vector<T> segtree;
+
+	T combine(const T &a, const T &b) { return std::min(a, b); }
+
+	void set(int ind, T val, int at, int at_left, int at_right) {
+		if (at_right - at_left == 1) {
+			segtree[at] = val;
+			return;
+		}
+		int mid = (at_left + at_right) / 2;
+		if (ind < mid) {
+			set(ind, val, 2 * at + 1, at_left, mid);
```

Comment:
usually we do the node ordering a bit different

root node is 1, left child is v * 2, right child is v * 2 + 1

again, slightly arbitrary, but that's just the norm & I think its nicer this way